oak_runtime/src/node/http/client.rs

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    /// Process an HTTP invocation for an external HTTP service.
+    async fn process_invocation(
+        &mut self,
+        runtime: &RuntimeProxy,
+        invocation: &Invocation,
+    ) -> Result<(), OakError> {
+        // TODO(#1717): Update HTTP client metrics
+
+        // Define a closure to ease logging the errors
+        let send_error = |code, msg| {
+            invocation.send_error(code, msg, runtime);
+        };
+
+        // Receive a request from the invocation channel.
+        let request = invocation.receive_request(&runtime).map_err(|error| {
+            send_error(
+                http::StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to read request",
+            );
+            error!(
+                "Couldn't read HTTP request from the invocation: {:?}",
+                error
+            );
+            error
+        })?;
+        debug!("Incoming HTTP request: {:?}", request);
+
+        let mut hyper_request_builder = hyper::Request::builder()
+            .method(&request.method[..])
+            .uri(request.uri);
+
+        if let Some(headers) = request.headers {
+            for (header_name, header_value) in headers.into_iter() {
+                hyper_request_builder = hyper_request_builder.header(header_name, header_value);
+            }
+        }
+        let hyper_request = hyper_request_builder
+            .body(hyper::Body::from(request.body))
+            .map_err(|err| {
+                warn!("Error when processing the request: {:?}", err);
+                OakError::OakStatus(OakStatus::ErrInternal)
+            })?;
+
+        // Forward the request to the external HTTP service and wait for the response.
+        let resp = self
+            .http_client
+            .request(hyper_request)
+            .await
+            .map_err(|error| {
+                error!("Error while awaiting response: {}", error);
+                // TO-DO: Retry instead?
+                send_error(
+                    http::StatusCode::INTERNAL_SERVER_ERROR,
+                    "Error while awaiting response",
+                );
+                OakStatus::ErrInternal
+            })?;
+
+        let response_handler = ResponseHandler::new(runtime.clone(), resp, invocation);
+        response_handler.handle().await
+    }
+}
+
+struct ResponseHandler<'a> {
+    runtime: RuntimeProxy,
+    response: http::Response<hyper::Body>,
+    invocation: &'a Invocation,
+}
+
+impl<'a> ResponseHandler<'a> {
+    fn new(
+        runtime: RuntimeProxy,
+        response: http::Response<hyper::Body>,
+        invocation: &'a Invocation,
+    ) -> Self {
+        ResponseHandler {
+            runtime,
+            response,
+            invocation,
+        }
+    }
+
+    async fn handle(&self) -> Result<(), OakError> {
+        let response = HttpResponse {
+            body: vec![],
+            status: 400,
+            headers: None,
+        };
```

Comment:
What does this response represent? Why are all its fields hardcoded? I guess this is not the real method that sends responses back over the invocation channel, or am I missing something?

Reply:
Oops! Thanks for catching it! Fixed. I'll add more checks to the test too.